torch/csrc/jit/codegen/cuda/runtime/tensor_core.cu

```diff
@@ -0,0 +1,410 @@
+
+// Arch guard, Volta + only
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 700))
+
+namespace mma_instruction {
+
+// MMA instruction wrappers (sm_70+):
+// The instruction wrappers below are quarter-warp macros, which currently
+// nvfuser
+//  doesn't explicitly model. So they are currently only meant to be
+//  used as building blocks in warp level mma macros
+
+//  8x8x4 mma instruction, per quarter warp (8 threads), fp16 accumulate
+//  per thread register:
+//   A[4] x B[4] -> C[8]
+__device__ inline void mmaM8n8k4(__half* C, __half* A, __half* B) {
+  unsigned const* _A = reinterpret_cast<unsigned const*>(A);
+  unsigned const* _B = reinterpret_cast<unsigned const*>(B);
+  unsigned* _C = reinterpret_cast<unsigned*>(C);
+
+  asm("mma.sync.aligned.m8n8k4.row.row.f16.f16.f16.f16 {%0,%1,%2,%3}, {%4,%5}, {%6,%7}, {%8,%9,%10,%11};\n"
+      : "=r"(_C[0]), "=r"(_C[1]), "=r"(_C[2]), "=r"(_C[3])
+      : "r"(_A[0]),
+        "r"(_A[1]),
+        "r"(_B[0]),
+        "r"(_B[1]),
+        "r"(_C[0]),
+        "r"(_C[1]),
+        "r"(_C[2]),
+        "r"(_C[3]));
+}
+
+//  8x8x4 mma instruction, per quarter warp (8 threads), fp32 accumulate
+//  per thread register:
+//   A[4] x B[4] -> C[8]
+__device__ inline void mmaM8n8k4(float* C, __half* A, __half* B) {
```

Comment:
Is there something we can pass in here that's encapsulating. I don't like the idea of passing direct pointers, I'd probably even prefer to pass in uint4 or uint8's, but something like an array class would be best if possible (I thought there was something like this we could use).

^^ for the rest of the functions in this namespace